### PR TITLE
Specify test dependencies

### PR DESCRIPTION
**Pull request by GeorgeNagel**
Feel free to close this if it's unwanted. The motivation behind this PR is to make it easier to contribute by being explicit about the test dependencies and adding instructions to the readme on how to run the test suite.

### CODE REVIEW

File: requirements_test.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-r requirements.txt
+
+pep8>=1.7.0
+nose>=1.3.7
```
Comment (jyegerlehner) [inline requirements_test.txt]: I ask out of ignorance: why doesn't librosa need to be there?

Comment (GeorgeNagel) [inline requirements_test.txt]: The line `-r requirements.txt` pulls in the base requirements (librosa and any others added in the future) from the requirements.txt file. I should probably add a clarifying comment.

Comment (ibab) [inline requirements_test.txt]: I think it's fine as it is 👍

**Comment by ibab** [comment]
Thanks for doing this!
I can merge this if you say you're finished with it.

**Comment by GeorgeNagel** [comment]
@ibab: Yes, I'm finished 😄

**Pull request merged**